post/getting-started-preact/index.md

```diff
@@ -0,0 +1,409 @@
+
+type Global = {
+    title: string;
+    description: string;
+};
+
+type Author = {
+    name: string;
+};
+
+type Page = {
+    title: string;
+    content: string;
+    slug: string;
+};
+
+type Post = {
+    image: string;
+    title: string;
+    author: Author;
+    content: string;
+    published_date: string;
+    slug: string;
+};
+
+type Schema = {
+    posts: Post[];
+    global: Global;
+    pages: Page[];
+};
+
+const directus = createDirectus<Schema>(
+    "https://directus-supabase.onrender.com"
+).with(rest());
+export default directus;
+```
+
+## Using Global Metadata
+
+In your Directus project, navigate to Settings -> Data Model and create a new collection called global. Under the Singleton option, select 'Treat as a single object', as this collection will have just a single entry containing global website metadata.
+
+Create two text input fields - one with the key of `title` and one `description`.
+
+Navigate to the `content` module and enter the `global` collection. Enter information in the title and description field and hit save.
+
+In `pages/index.jsx` file, add the following code to fetch the data from Directus and display it.
+
+```js
+import directus from "../utils/directus";
+import { readSingleton } from "@directus/sdk";
+import { useState } from "preact/compat";
+
+export function Home() {
+    const [global, setGlobal] = useState({});
+    const fetchData = async () => {
+        const global = await directus.request(readSingleton("global"));
+
+        setGlobal(global);
+    };
+    fetchData();
+
+    return (
+        <div>
+            Home Page
+            <h1>{global.title}</h1>
+        </div>
+    );
+}
+```
+
+### Make Collection Public
+
+Navigate to Settings -> Access Control -> Public and give Read access to the `global` collection.
+
+![Global Metadata](./images/global-metadata.png)
+
+## Setup Routing
+
+We will load Blog Post List, detail page, dynamic pages and a 404 page.
+
+```js
+import { render } from "preact";
+import { LocationProvider, Router, Route } from "preact-iso";
+
+import { NotFound } from "./pages/_404.jsx";
+import { BlogList } from "./pages/blog/list.jsx";
+import { Blog } from "./pages/blog/slug.jsx";
+import { Page } from "./pages/slug.jsx";
+import { Home } from "./pages/index.jsx";
+
+import { Header } from "./components/Header.jsx";
+import "./style.css";
+export function App() {
+    return (
+        <LocationProvider>
+            <Header />
+            <main>
+                <Router>
+                    <Route path="/" component={Home} />
+
+                    <Route path="/blog" component={BlogList} />
+                    <Route path="/404" component={NotFound} />
+                    <Route path="/blog/:slug" component={Blog} />
+                    <Route path="/:slug" component={Page} />
+                </Router>
+            </main>
+        </LocationProvider>
+    );
+}
+render(<App />, document.getElementById("app"));
+```
```

Comment:
I get that you're trying to centralize the work here, but please can you make this only for the home, and at each step: :slug, 404, /blog, /blog/:slug, add the import and component

Reply:
@phazonoverload Can you explain with above example, I am not able to understand what is the problem in this above code?

Reply:
Do less upfront - just do the Home Route, and then when you move on to doing dynamic pages, then as part of the setup add the Page route here.